src/Type.cc

```diff
@@ -1591,6 +1591,11 @@ void EnumType::CheckAndAddName(const string& module_name, const char* name, zeek
     if ( vals.find(val) == vals.end() )
         vals[val] = make_intrusive<EnumVal>(IntrusivePtr{NewRef{}, this}, val);
 
+    if ( ! id->HasVal() )
+        id->SetVal(vals[val]);
+    else if ( id->GetVal()->AsEnum() != val )
+        reporter->InternalError("inconsistent enum integer value");
```

Comment:
nit: Can this include the name it's trying to add?

```suggestion
        reporter->InternalError("inconsistent enum integer value for '%s'", fullname.c_str());
```